### DISCUSSION TITLE
Sending something compressed

### DISCUSSION THREAD

**Discussion by carlosvdr** [paritytech/jsonrpsee#1379] [Q&A]
Is there a way to compress something through this crate, im trying to compress at request for upload, want to use http layering but seems cant modify headers in here, what should be the path?

**Comment by niklasad1** [comment]
Hey @carlosvdr 

If you are using the HTTP client you can inject your own http middleware/tower middleware for that. Have a look [at this example](https://github.com/paritytech/jsonrpsee/blob/master/examples/examples/http.rs#L48-#L65)

**Reply by carlosvdr** [reply]
Hey I did tried doing so, but there where many dependencies issues, between tower and that example, had to use your main branch not a deployed one but that had other dependencies issues at the end.
Perhaps I can show you about it?

**Reply by niklasad1** [reply]
good ahead and paste your code in the Q&A and we might be able to help out
